report/src/main/kotlin/com/hartwig/actin/report/pdf/tables/molecular/OrangeMolecularRecordGenerator.kt

```diff
@@ -67,10 +69,36 @@ class OrangeMolecularRecordGenerator(
         return if (molecular.hasSufficientQuality) {
             listOf(
                 PredictedTumorOriginGenerator(molecular),
-                MolecularDriversGenerator(molecular, evaluated, trials)
+                MolecularDriversGenerator(
+                    molecular.copy(drivers = filterDriversByDriverLikelihood(molecular.drivers, true)),
+                    evaluated,
+                    trials,
+                    "Key drivers"
+                ),
+                MolecularDriversGenerator(
+                    molecular.copy(drivers = filterDriversByDriverLikelihood(molecular.drivers, false)),
+                    evaluated,
+                    trials,
+                    "Other events"
+                )
             )
         } else emptyList()
     }
+
+    fun filterDriversByDriverLikelihood(drivers: Drivers, useHighDrivers: Boolean): Drivers {
+        fun <T> filterByDriverLikelihood(items: List<T>, getDriverLikelihood: (T) -> DriverLikelihood?): List<T> =
+            if (useHighDrivers) items.filter { getDriverLikelihood(it) == DriverLikelihood.HIGH }
+            else items.filter { getDriverLikelihood(it) != DriverLikelihood.HIGH }
+
+        return Drivers(
+            variants = filterByDriverLikelihood(drivers.variants) { it.driverLikelihood },
+            copyNumbers = filterByDriverLikelihood(drivers.copyNumbers) { it.driverLikelihood },
+            homozygousDisruptions = filterByDriverLikelihood(drivers.homozygousDisruptions) { it.driverLikelihood },
+            disruptions = filterByDriverLikelihood(drivers.disruptions) { it.driverLikelihood },
+            fusions = filterByDriverLikelihood(drivers.fusions) { it.driverLikelihood },
+            viruses = filterByDriverLikelihood(drivers.viruses) { it.driverLikelihood }
+        )
+    }
```

Comment:
I was looking for a simpler way to do this using `.partition` since you should be able to filter with one pass, but the structure of `Drivers` makes that difficult. An extension function can help though:

```suggestion
    private fun <T: Driver> List<T>.matchingLikelihood(useHighDrivers: Boolean) =
        this.filter { (it.driverLikelihood == DriverLikelihood.HIGH) == useHighDrivers }
        
    fun filterDriversByDriverLikelihood(drivers: Drivers, useHighDrivers: Boolean): Drivers {
        return with(drivers) {
            Drivers(
                variants = variants.matchingLikelihood(useHighDrivers),
                copyNumbers = copyNumbers.matchingLikelihood(useHighDrivers),
                homozygousDisruptions = homozygousDisruptions.matchingLikelihood(useHighDrivers),
                disruptions = disruptions.matchingLikelihood(useHighDrivers),
                fusions = fusions.matchingLikelihood(useHighDrivers),
                viruses = viruses.matchingLikelihood(useHighDrivers)
            )
        }
    }
```